### PR TITLE
Check output

### PR DESCRIPTION
**Pull request by u2**
Many duplicated code :cry: 
@classicalliu , @ashchan

### CODE REVIEW

**Comment by ashchan** [comment]
@u2 We have these primitive hash code duplicated as we want to use each script as an isolated one without depending on others.

Last we we've made some changes and update all from using SHA3 to Blake2B. This resulted this PR having conflicts. Could you rebase or merge to update this PR? Thanks!

**Comment by u2** [comment]
updated

File: bitcoin/sighash_multiple.rb [hunk 1]
```diff
@@ -16,6 +16,8 @@ def hex_to_bin(s)
   [s].pack("H*")
 end
 
+output_index_error = "Output index error!".freeze
```
Comment (xxuejie) [inline bitcoin/sighash_multiple.rb]: Can we use all upper case letters for constants?

Comment (u2) [reply bitcoin/sighash_multiple.rb]: ok

**Review by xxuejie — APPROVED** [review]

**Pull request merged**